exporters/otlp/logs/src/test/java/io/opentelemetry/exporter/otlp/logs/OtlpGrpcLogsExporterTest.java

```diff
@@ -8,77 +8,70 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatCode;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
-import static org.awaitility.Awaitility.await;
 
-import com.google.common.io.Closer;
-import io.github.netmikey.logunit.api.LogCapturer;
-import io.grpc.ManagedChannel;
-import io.grpc.Server;
-import io.grpc.Status;
-import io.grpc.inprocess.InProcessChannelBuilder;
-import io.grpc.inprocess.InProcessServerBuilder;
-import io.grpc.stub.StreamObserver;
 import io.opentelemetry.api.common.Attributes;
-import io.opentelemetry.exporter.otlp.internal.RetryPolicy;
-import io.opentelemetry.exporter.otlp.internal.grpc.DefaultGrpcExporter;
-import io.opentelemetry.exporter.otlp.internal.grpc.DefaultGrpcExporterBuilder;
+import io.opentelemetry.exporter.otlp.internal.Marshaler;
+import io.opentelemetry.exporter.otlp.internal.grpc.OkHttpGrpcExporterBuilder;
 import io.opentelemetry.exporter.otlp.internal.logs.ResourceLogsMarshaler;
-import io.opentelemetry.proto.collector.logs.v1.ExportLogsServiceRequest;
-import io.opentelemetry.proto.collector.logs.v1.ExportLogsServiceResponse;
-import io.opentelemetry.proto.collector.logs.v1.LogsServiceGrpc;
+import io.opentelemetry.exporter.otlp.testing.internal.AbstractGrpcTelemetryExporterTest;
 import io.opentelemetry.proto.logs.v1.ResourceLogs;
 import io.opentelemetry.sdk.common.CompletableResultCode;
 import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
 import io.opentelemetry.sdk.logs.data.LogData;
 import io.opentelemetry.sdk.logs.data.LogDataBuilder;
 import io.opentelemetry.sdk.logs.data.Severity;
 import io.opentelemetry.sdk.resources.Resource;
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
 import java.time.Duration;
 import java.time.Instant;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
-import java.util.stream.Collectors;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.RegisterExtension;
-import org.slf4j.event.Level;
-import org.slf4j.event.LoggingEvent;
 
-class OtlpGrpcLogsExporterTest {
+class OtlpGrpcLogsExporterTest
+    extends AbstractGrpcTelemetryExporterTest<LogData, ResourceLogs, OtlpGrpcLogExporter> {
```

Comment:
I'm very very leery about creating a test class hierarchy. This inevitably leads to tangled, unmaintainable, incomprehensible test code. Can we use some static helper methods that can be shared, rather than creating a class hierarchy?

Reply:
I think a big part of keeping the test code comprehensible if we go down the class hierarchy route is being meticulous about limiting test scope. 

For example, this class asserts common export patterns shared between the exporters.  Subclasses should only implement the required abstract methods and test nothing else. If other verifications are needed, create a new test class. So if following this advice, the subclass `OtlpGrpcLogsExporterTest` would  move the config validation tests to a different class.

Reply:
I think moving the config tests out makes sense. I agree @jkwatson with the sentiment of not creating spaghetti, but in this case the tests are 100% the same, so I think it's appropriate here